src/Adapter/Cart/QueryHandler/GetCartInformationHandler.php

```diff
@@ -238,4 +272,120 @@ private function fetchCartDeliveryOptions(array $deliveryOptionsByAddress, int $
         //make sure array is not associative
         return array_values($deliveryOptions);
     }
+
+    /**
+     * @param array $legacySummary
+     * @param Currency $currency
+     * @param Cart $cart
+     *
+     * @return CartInformation\CartSummary
+     *
+     * @throws LocalizationException
+     */
+    private function extractSummaryFromLegacySummary(array $legacySummary, Currency $currency, Cart $cart): CartSummary
+    {
+        $cartId = (int) $cart->id;
+        $discount = $this->locale->formatPrice($legacySummary['total_discounts_tax_exc'], $currency->iso_code);
+
+        if (0 !== (int) $legacySummary['total_discounts_tax_exc']) {
+            $discount = '-' . $discount;
+        }
+
+        $orderMessage = '';
+        if ($message = Message::getMessageByCartId($cartId)) {
+            $orderMessage = $message['message'];
+        }
+
+        return new CartSummary(
+            $this->locale->formatPrice($legacySummary['total_products'], $currency->iso_code),
+            $discount,
+            $this->locale->formatPrice($legacySummary['total_shipping_tax_exc'], $currency->iso_code),
+            $this->locale->formatPrice($legacySummary['total_tax'], $currency->iso_code),
+            $this->locale->formatPrice($legacySummary['total_price'], $currency->iso_code),
+            $this->locale->formatPrice($legacySummary['total_price_without_tax'], $currency->iso_code),
+            $orderMessage,
+            $this->contextLink->getPageLink(
+                'order',
+                false,
+                (int) $cart->id_lang,
+        'step=3&recover_cart=' . $cart->id . '&token_cart=' . md5(_COOKIE_KEY_ . 'recover_cart_' . $cart->id)
```

Comment:
Can we use https://www.php.net/manual/en/function.http-build-query.php instead ?